CIP-????/README.md

```diff
@@ -0,0 +1,119 @@
+---
+CIP: ?
+Title: Authenticated Web3 HTTP requests
+Category: Tools
+Status: Proposed
+Authors:
+    - Juan Salvador Magán Valero <jmaganvalero@gmail.com>
+Implementors: []
+Discussions:
+    - https://github.com/cardano-foundation/cips/pulls/?
+Created: 2022-12-27
+License: CC-BY-4.0
+---
+
+# CIP-XXXX: Authenticated Web3 HTTP requests
+
+## Abstract
+
+The cardano wallets have the ability to sign arbitrary pieces of data as we can see in the [Message signing CIP-0008](/CIP-0008/README.md). All wallets implement the method ```api.signData(addr: Address, payload: Bytes): Promise<DataSignature>``` defined in [Cardano dApp-Wallet Web Bridge CIP-0030](/CIP-0030/README.md).
+
+An important use case for web3 sites is calling HTTP endpoints using authenticated requests signed by the private keys of the users' wallets. In this way, the servers can perform actions using onchain and offchain data. 
+
+
+## Motivation
+
+dApps generate arbritary payloads as byte arrays. These payloads are signed and included in the protected headers of the signatures. The wallets are responsible for showing the payload data to the user, who will proceed to sign or reject the payload. It's a common practice to encode a string or a JSON string but there isn't any standard for the way to construct and to show this data.
+
+The current implementations for web3 applications use static strings. This is dangerous because if a bad actor intercepts the signed message then it can be used in a replay attack by the bad actor. That's why it is very important to produce a dynamic payload rather a static string.
+
+Another problem with the current approach is how the wallets show the information contained in the payload. The payload is a enconded byte array and it could contain anything. If Alice want to call an endpoint and Bob has the ability to change the message before Alice gets it. Alice must be notified somehow that she is signing a potentially malicious payload. A simple hex-encoded representation of the payload isn't enough to ensure a safe interaction.
+
+## Specification
+
+This specification involves multiple parties. We split this specification in three parts: server processing, payload structure and wallet specification.
```

Comment:
It isn't clear to me what "the server" refers to in this proposal. I think the proposal would deserve a bit of an introduction regarding each actor involved and their role.

Reply:
I added dApp Server and the ASCII diagram. I believe now it's clearer.

Reply:
@rphair I added some definition before the ASCII diagram:

> This specification involves multiple parties: Wallet/Client, dApp Server and Blockchain. 
> 
> 1. **Wallet/Client**: The Wallet/Client is responsible for managing the user's cryptographic keys. Anyone can create a wallet using the CIP-0030 API interface, but it may produce invalid or malicious data sent to the dApp. This CIP aims to validate the ownership and veracity of the data provided by wallets. Additionally, it establishes guidelines for mitigating common wallet attacks, improving security for user interaction.
> 
> 2. **dApp Server**: The dApp Server represents the server-side infraestructure that supports decentralized applications (dApps). It communicates with the blockchain to retrieve stored data and validate wallet status. It must enforce minimum payload requirements to ensure authenticity and protect users from malicious actors.
> 
> 3. **Blockchain**: The Blockchain is the underlying distributed ledger technology that forms the foundation of decentralized systems. It is a decentralized and immutable ledger that securely records all transactions and data in a chronological and transparent manner. The Blockchain can be utilized for authentication, providing user identity, and for authorization, tracking user history and current status.

I believe this gives more background and context to each actor.